src/main/perl/RuleBasedEditor.pm

```diff
@@ -0,0 +1,1023 @@
+# ${license-info}
+# ${developer-info}
+# ${author-info}
+# ${build-info}
+#
+#
+# This module implements a rule-based editor that is used to modify the content
+# of an existing file without taking care of the whole file. Each rule
+# driving the edition process is applied to all matching lines. The input for
+# updating the file is the Quattor configuration and conditions can be defined
+# based on the contents of this configuration.
+#
+# This module is a subclass of the CAF FileEditor: it extends the base methods of 
+# the CAF FileEditor. In addition to the constructor it has only one public method.
+# The methods provided in this module can be used at the same time as the
+# base methods of the FileEditor.
+#
+#
+#######################################################################
+
+package CAF::RuleBasedEditor;
+
+use strict;
+use warnings;
+use vars qw(@ISA $EC);
+$EC=LC::Exception::Context->new->will_store_all;
+
+use parent qw(CAF::FileEditor Exporter);
+
+use EDG::WP4::CCM::Element;
+
+use Readonly;
+
+use Encode qw(encode_utf8);
+
+local(*DTA);
+
+# Constants from FileEditor
+use CAF::FileEditor qw(BEGINNING_OF_FILE ENDING_OF_FILE);
+
+
+#########################################################
```

Comment:
pod style documentation if you wan this rendered as part of the documentation on the website (and i think you do :smile: )